packages/plugin-team/src/decrypter.js

```diff
@@ -0,0 +1,64 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {base64} from '@ciscospark/common';
+import {Decrypter} from '@ciscospark/plugin-conversation';
+
+/**
+ * Decrypts a team
+ * @param {Object} team
+ * @param {Encryption~Key|string} key
+ * @param {Object} options
+ * @param {Object} parentObject
+ * @returns {Promise} Resolves with the decrypted team
+ * @private
+ */
+Decrypter.prototype._decryptTeam = function _decryptTeam(team, key, options, parentObject) {
+  let promises = [];
+  if (team.conversations.items) {
+    promises = team.conversations.items.map((item) => this.decryptObject(item, options));
+  }
+
+  const usableKey = key || team.encryptionKeyUrl;
+
+  if (usableKey && team.displayName) {
+    try {
+      const parts = team.displayName.split(`.`);
+      if (parts.length === 5) {
+        JSON.parse(base64.decode(parts[0]));
+
+        // Much like a conversation, a team can be the object of update, add or leave activities.
+        // If this is the case we measure decryption of parent object
+        this.spark.encryption.metrics.startDecryptionMetrics(parentObject || team);
```

Comment:
encryption metrics doesn't exist at the moment; feel free to remove the metrics lines in here.